### PR TITLE
Decouple parser and storage

### PR DESCRIPTION
**Pull request by ruivieira**
At the moment the parsing of data (read from an abstract `Storage`) was done in the data provider.

This PR adds a new abstraction `DataParser` in which new data formats can be implemented.
This allows combinations of storage and data (e.g. JSON in MinIO or CSV in PVC) to be implemented separately as a new format will be available for all storage types.

See #29 .

### CODE REVIEW

File: src/main/java/org/kie/trustyai/service/data/DataSource.java [hunk 1]
```diff
@@ -1,66 +1,60 @@
...
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
+import io.quarkus.arc.DefaultBean;
```
Comment (danielezonca) [inline src/main/java/org/kie/trustyai/service/data/DataSource.java]: Unused import?

File: pom.xml [hunk 2]
```diff
@@ -88,6 +88,19 @@
...
+        <version>3.0.5</version>
+        <executions>
+          <execution>
+            <id>make-index</id>
+            <goals>
+              <goal>jandex</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (danielezonca) [inline pom.xml]: Is it this necessary? It seems strange to me 🤔

File: src/main/java/org/kie/trustyai/service/data/DataSource.java [hunk 3]
```diff
@@ -1,66 +1,60 @@
...
 @Singleton
-public class DataParser {
-    private static final Logger LOG = Logger.getLogger(DataParser.class);
-    private static final Charset UTF8 = StandardCharsets.UTF_8;
-    @Inject MinioStorage storage;
+public class DataSource {
+    private static final Logger LOG = Logger.getLogger(DataSource.class);
+
+    @Inject
+    Instance<Storage> storage;
```
Comment (danielezonca) [inline src/main/java/org/kie/trustyai/service/data/DataSource.java]: You should be able to directly inject the `Storage` and not the `Instance<Storage>`.
Btw the `get()` method can fail so if we cannot inject `Storage` directly we should at least catch the runtime exception

Comment (ruivieira) [reply src/main/java/org/kie/trustyai/service/data/DataSource.java]: Good point. The `Instance<T>` is now removed.

File: pom.xml (hunk 2 above)
Comment (ruivieira) [reply pom.xml]: Letfover from experiment. Thanks!

**Review by danielezonca — APPROVED** [review]
LGTM 👍

**Pull request merged**